tests/api/db/test_artifacts.py

```diff
@@ -883,6 +904,78 @@ def test_migrate_artifacts_to_v2(db: DBInterface, db_session: Session):
             )
 
 
+def test_migrate_artifact_v2_tag(db: DBInterface, db_session: Session):
+    artifact_key = "artifact1"
+    artifact_uid = "uid1"
+    artifact_tag = "artifact-tag-1"
+    project = "project1"
+
+    # create project
+    db.create_project(
+        db_session,
+        mlrun.common.schemas.Project(
+            metadata=mlrun.common.schemas.ProjectMetadata(
+                name=project,
+            ),
+            spec=mlrun.common.schemas.ProjectSpec(description="some-description"),
+        ),
+    )
+
+    # create an artifact in the old format
+    artifact_body = _generate_artifact(artifact_key, artifact_uid, "artifact")
+    artifact_body["metadata"]["key"] = artifact_key
+    artifact_body["metadata"]["iter"] = 2
+    artifact_body["metadata"]["project"] = project
+    db.store_artifact_v1(
+        db_session,
+        artifact_key,
+        artifact_body,
+        artifact_uid,
+        project=project,
+        tag=artifact_tag,
+    )
+
+    query_all = db._query(
+        db_session,
+        server.api.db.sqldb.models.Artifact,
+    )
+    old_artifacts = query_all.all()
+    assert len(old_artifacts) == 1
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # change the state file path to the temp directory for the test only
+        mlrun.config.config.artifacts.artifact_migration_state_file_path = (
+            temp_dir + "/_artifact_migration_state.json"
+        )
+
+        # perform the migration
+        server.api.initial_data._migrate_artifacts_table_v2(db, db_session)
+
+    # validate the migration succeeded
+    query_all = db._query(
+        db_session,
+        server.api.db.sqldb.models.ArtifactV2,
+    )
+    new_artifacts = query_all.all()
+    assert len(new_artifacts) == 1
```

Comment:
```suggestion
    new_artifact = query_all.one()
```
`one()` asserts if query results in not exactly a single result. It also returns the object, so you don't need `new_artifacts[0]` later.

Reply:
Nice :)